aten/src/ATen/core/op_registration/README.md

```diff
@@ -0,0 +1,239 @@
+
+The kernel function can take and return dicts by using `torch::Dict<Key, Value>`. `Key` must be `int64_t`, `std::string`, `double` or `bool`, and `Value` must be from the list of supported types above excluding `at::Scalar`.
+
+When taken as input, any of these types can be taken by value (i.e. `Tensor`) or by const-reference (i.e. `const Tensor&`). We recommend taking all arguments by value, even Tensors. They will be moved in, so there is no performance overhead.
+
+If you need another type, it might work but not be officially supported (yet). Please reach out to Sebastian Messmer and we'll see what we can do.
+
+### Overloads
+
+When multiple kernels are registered for the same operator, they must have the same schema or registration will fail.
+*Note: This also includes schema properties like annotations or default arguments. If one kernel specifies a schema with annotations or a default argument, all kernels for this operator must do this. Schemas automatically inferred from kernel functions will not have annotations or default arguments. This means to use annotations or default arguments, all kernels for this operator must explicitly specify the schema.*
+
+If you want to reuse the same operator name for a different schema, you can use overloads. Overloads must be named and the name is appended to the operator name after a dot:
+
+```
+namespace {
+  Tensor my_kernel_cpu_1(const Tensor& a) {...}
+  Tensor my_kernel_cpu_2(const Tensor& a, const Tensor& b) {...}
+}
+
+static auto registry = torch::RegisterOperators()
+   .op("my_namespace::my_op.overload1(Tensor a) -> Tensor",
+       torch::RegisterOperators::options()
+         .kernel<decltype(my_kernel_cpu_1), &my_kernel_cpu>(CPUTensorId()))
+   .op("my_namespace::my_op.overload2(Tensor a, Tensor b) -> Tensor",
+       torch::RegisterOperators::options()
+         .kernel<decltype(my_kernel_cpu_2), &my_kernel_cpu>(CPUTensorId()));
+```
+
+Kernels registered for the same overload must have exactly matching schemas, but kernels registered for different overloads are allowed to have different schemas. This also works when different overloads come from different shared libraries.
+
+### Schema-only operators
+
+You can register an operator without a kernel:
+
+```
+static auto registry = torch::RegisterOperators()
+   .op("my_namespace::my_op(Tensor a, Tensor b) -> Tensor");
+```
+
+In this case, you must explicitly specify the full schema and you must not specify a dispatch key.
+This is useful to define the interface of an operator when you don't know a kernel yet. As mentioned above in the “Overloads” section, you will get an error if any kernel registered for this operator has a mismatching signature.
+
+## Calling custom operators
+
+### From PyTorch/JIT
+
+All registered operators are automatically available to PyTorch and JIT under `torch.ops.XXX`. If your operator was `my_namespace::my_op`, you can call it from python or JIT using `torch.ops.my_namespace.my_op(a, b)`.
```

Comment:
If we can create a small example, e.g., code snippet/test file, it will be easier for the reader.